prow/repoowners/repoowners.go

```diff
@@ -371,35 +376,42 @@ func (o *RepoOwners) walkFunc(path string, info os.FileInfo, err error) error {
 		return nil
 	}
 
+	if err := ParseOwners(path, o); err != nil {
+		log.WithError(err).Errorf("Error occured parsing the OWNERS file: %v.", err)
+		return nil
+	}
+
+	return nil
+}
+
+// ParseOwners will open and unmarshal OWNERS into a RepoOwners struct.
+// Returns an error if the file cannot be read, unmarshalled or its config applied.
+func ParseOwners(path string, o *RepoOwners) error {
```

Comment:
This doesn't need to take `RepoOwners` as a parameter. It should instead parse/coerce the OWNERS file into a `fullConfig` that is returned. In the walkFunc we can actually apply the parsed OWNERS file data to the RepoOwners struct.
Then the plugin won't need to create a RepoOwners struct that won't be used.